pkg/dataplane/server/listener.go

```diff
@@ -0,0 +1,89 @@
+package server
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/clusterlink-net/clusterlink/pkg/controlplane/api"
+	dpapi "github.com/clusterlink-net/clusterlink/pkg/dataplane/api"
+)
+
+// DeleteListener deletes the listener to an imported service
+func (d *Dataplane) DeleteListener(name string) {
+	d.listenerEnd[name] <- true
+}
+
+// CreateListener starts a listener to an imported service
+func (d *Dataplane) CreateListener(name, ip string, port uint32) {
+	listenTarget := ip + ":" + strconv.Itoa(int(port))
+	d.listenerEnd[name] = make(chan bool)
+	d.logger.Infof("Starting a listener for imported service %s at %s.", name, listenTarget)
+	acceptor, err := net.Listen("tcp", listenTarget)
+	if err != nil {
+		d.logger.Infof("Error Listen to port %v", err)
+		return
+	}
+	go func() {
+		if err = d.serveEgressConnections(name, acceptor); err != nil {
+			d.logger.Errorf("Failed to serve egress connection on %s: %+v.", listenTarget, err)
+		}
+	}()
+	<-d.listenerEnd[name]
+	acceptor.Close()
+}
+
+func (d *Dataplane) serveEgressConnections(name string, listener net.Listener) error {
+	for {
+		d.logger.Infof("Serving for imported service %s at %s.", name, listener.Addr())
+		conn, err := listener.Accept()
+		if err != nil {
+			d.logger.Error("Failed to accept egress connection", err)
+			return err
+		}
+
+		d.logger.Debugf("Received an egress connection at listener for imported service %s from %s.", name, conn.RemoteAddr().String())
+		d.logger.Debugf("Connection: %+v.", conn)
+
+		targetPeer, accessToken, err := d.getEgressAuth(name, strings.Split(conn.RemoteAddr().String(), ":")[0])
+		if err != nil {
+			d.logger.Error("Failed egress authorization", err)
```

Comment:
IMO If the error is due to a policy, this should be recorded at info level, not error level.

Reply:
Yes, you are right. Changed to info.